### PR TITLE
Cors feature

### LINKED ISSUE #1: CORS required
CORS are required to properly and securely connect with a frontend side. Keep in mind to implement it.

### PR DESCRIPTION
**Pull request by MateNaw**

### CODE REVIEW

File: YellowNotes.Api/Startup.cs [hunk 1]
```diff
@@ -3,9 +3,16 @@
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Hosting;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
 
 namespace YellowNotes.Api
 {
+    public class AllowedHosts
```
Comment (lkurzyniec) [inline YellowNotes.Api/Startup.cs]: separate file

File: YellowNotes.Api/Startup.cs [hunk 2]
```diff
@@ -17,6 +24,29 @@ public Startup(IConfiguration configuration)
 
         public void ConfigureServices(IServiceCollection services)
         {
+            AllowedHosts allowed;
+            using (FileStream fs = File.OpenRead("appsettngs.json"))
+                {
+                allowed = await JsonSerializer.DeserializeAsync<AllowedHosts>(fs);
+                }
```
Comment (lkurzyniec) [inline YellowNotes.Api/Startup.cs]: definitely not in that way, please take a look at how your friends did it.

File: YellowNotes.Api/Startup.cs [hunk 3]
```diff
@@ -17,6 +24,29 @@ public Startup(IConfiguration configuration)
...
+
+            services.AddCors(options =>
+            {
+            options.AddDefaultPolicy(
+                builder =>
+                {
+                   
+                    builder.WithOrigins(allowed.allowedHost);
```
Comment (lkurzyniec) [inline YellowNotes.Api/Startup.cs]: do we need two policies?

**Review by lkurzyniec — CHANGES_REQUESTED** [review]

File: YellowNotes.Api/Startup.cs (hunk 1 above)
Comment (MateNaw) [reply YellowNotes.Api/Startup.cs]: I did it without that class right now is that ok?

File: YellowNotes.Api/Startup.cs (hunk 2 above)
Comment (MateNaw) [reply YellowNotes.Api/Startup.cs]: refactored it into working one

File: YellowNotes.Api/Startup.cs (hunk 3 above)
Comment (MateNaw) [reply YellowNotes.Api/Startup.cs]: Removed it right now

**Review by wkud — DISMISSED** [review]

File: YellowNotes.Api/appsettings.json [hunk 4]
```diff
@@ -6,5 +6,15 @@
       "Microsoft.Hosting.Lifetime": "Information"
     }
   },
-  "AllowedHosts": "*"
+  "CORS-Settings": {
```
Comment (lkurzyniec) [inline YellowNotes.Api/appsettings.json]: Here should be also PascalCase so `CorsSettings`

File: YellowNotes.Api/Startup.cs [hunk 5]
```diff
@@ -21,7 +24,19 @@ public Startup(IConfiguration configuration)
         public IConfiguration Configuration { get; }
 
         public void ConfigureServices(IServiceCollection services)
-        {
+        {    
+            var allowedHosts = Configuration.GetSection("CORS-Settings:AllowedHosts").Get<string[]>();
```
Comment (lkurzyniec) [inline YellowNotes.Api/Startup.cs]: now is better :)

File: YellowNotes.Api/Startup.cs [hunk 6]
```diff
@@ -21,7 +24,19 @@ public Startup(IConfiguration configuration)
...
+            var allowedMethods = Configuration.GetSection("CORS-Settings:AllowedMethods").Get<string[]>();
+            services.AddCors(options =>
+            {
+            options.AddDefaultPolicy(
```
Comment (lkurzyniec) [inline YellowNotes.Api/Startup.cs]: please fix your indentation.

**Review by lkurzyniec — DISMISSED** [review]

**Review by lkurzyniec — DISMISSED** [review]
Now it looks ok. 

Remember about proper indentation.

File: YellowNotes.Api/Startup.cs [hunk 7]
```diff
@@ -1,10 +1,13 @@
+using System.Data;
 using System;
 using Microsoft.AspNetCore.Builder;
 using Microsoft.AspNetCore.Hosting;
 using Microsoft.EntityFrameworkCore;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Hosting;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
```
Comment (c1rcle) [inline YellowNotes.Api/Startup.cs]: These imports are not used, please remove them.

**Review by c1rcle — DISMISSED** [review]

**Review by c1rcle — APPROVED** [review]

**Pull request merged**